docs/architecture.md

```diff
@@ -0,0 +1,106 @@
+
+### Waiting until EL clients have started mining
+Once we have a network of EL nodes started, we block until all EL clients have a block number > 0 to ensure that they are in fact working. After the nodes have started mining, we're ready to move on to adding the CL client network.
+
+### Generating CL client data
+CL clients, like EL clients, also have genesis and config files that they need. We use [the same Docker image with tools for generating genesis data][ethereum-genesis-generator] to create the necessary CL files, we provide the genesis generation config using templates in [the `static_files` directory][static-files], and we store the generated output files in the Kurtosis enclave in the same way as the EL client genesis files. Like with EL nodes, CL genesis data information is tracked in [the `CLGenesisData` object](https://github.com/kurtosis-tech/eth2-merge-kurtosis-module/blob/master/kurtosis-module/impl/participant_network/cl/cl_client_context.go).
+
+The major difference with CL clients is that CL clients have validator keys, so we need to generate keys for the CL clients to use during validation. The generation happens using [the same genesis-generating Docker image][ethereum-genesis-generator], but via a call to [a different tool bundled inside the Docker image](https://github.com/protolambda/eth2-val-tools).
+
+### Starting CL clients
+Once CL genesis data and keys have been created, the CL client nodes are started via [the CL client launchers](https://github.com/kurtosis-tech/eth2-merge-kurtosis-module/tree/master/kurtosis-module/impl/participant_network/cl). Just as with EL clients:
+
+- CL client launchers implement [the `CLClientLauncher` interface](https://github.com/kurtosis-tech/eth2-merge-kurtosis-module/blob/master/kurtosis-module/impl/participant_network/cl/cl_client_launcher.go)
+- One CL client launcher exists per client type (e.g. [Nimbus](https://github.com/kurtosis-tech/eth2-merge-kurtosis-module/blob/master/kurtosis-module/impl/participant_network/cl/nimbus/nimbus_launcher.go), [Lighthouse](https://github.com/kurtosis-tech/eth2-merge-kurtosis-module/tree/master/kurtosis-module/impl/participant_network/cl/lighthouse))
+- Launched CL node information is tracked in [a `CLClientContext` object](https://github.com/kurtosis-tech/eth2-merge-kurtosis-module/blob/master/kurtosis-module/impl/participant_network/cl/cl_client_context.go)
+
+There are only two major difference between CL client and EL client launchers. First, the `CLClientLauncher.Launch` method also consumes an `ELClientContext`, because each CL client is connected in a 1:1 relationship with an EL client. Second, because CL clients have keys, the keystore files are passed in to the `Launch` function as well.
+
+Auxiliary Services
+------------------
+After the Ethereum network is up and running, this module starts several auxiliary Docker containers to make it easier to work with the Ethereum network. At time of writing, these are:
```

Comment:
i suggest s/Docker containers/services to hide impl. I know Ethereum uses Docker right now - but they might be using K8s/Pods at some point and this might imply that Kurtosis is only on Docker.

Reply:
or just containers

Reply:
great idea; fixt